adaptive_memory_v4.0.py

```
@@ -0,0 +1,1769 @@
+                clusters.append(cluster)
+                logger.info(f"Found cluster with {len(cluster)} memories (similarity >= {self.valves.summarization_similarity_threshold})")
+
+        logger.info(f"Found {len(clusters)} clusters ready for summarization")
+
+        # 4. Summarize Clusters
+        for cluster_indices in clusters:
+            try:
+                cluster_memories = [memories[i] for i in cluster_indices]
+                cluster_text = "\n".join([f"- {m.content}" for m in cluster_memories])
+
+                prompt = self.valves.summarization_memory_prompt + f"\n\n{cluster_text}"
```

Ruff (0.14.10)

Local variable `prompt` is assigned to but never used

Remove assignment to unused variable `prompt`

(F841)